startup/19-pseudomotors.py

```diff
@@ -0,0 +1,66 @@
+print(f'Loading {__file__}...')
+
+import math
+from ophyd import (Device, PVPositionerPC, EpicsMotor, Signal, EpicsSignal,
+                   EpicsSignalRO, Component as Cpt, FormattedComponent as FCpt,
+                   PseudoSingle, PseudoPositioner,
+                   )
+from ophyd.pseudopos import (real_position_argument,
+                             pseudo_position_argument)
+from hxntools.device import NamedDevice #qustion for Ken
+
+class FineSampleLabX(PseudoPositioner, NamedDevice):
+    '''Pseudo positioner definition for zoneplate fine sample positioner
+    with angular correction
+    (-1): is due to the reverse z direction
+    '''
+    # pseudo axes
+    lab_x = Cpt(PseudoSingle)
+    lab_z = Cpt(PseudoSingle)
+
+    # real axes
+    real_topx = Cpt(EpicsMotor, 'XF}Mtr', name='hf_stage_topx')
+    real_topz = Cpt(EpicsMotor, 'ZF}Mtr', name='hf_stage_topz')
+    real_theta = Cpt(EpicsMotor, 'Rot}Mtr', name='hf_stage_th')
+
+    # configuration settings
+    _theta0 = Cpt(Signal, value=-45.0, doc='theta offset')
+
+    def __init__(self, prefix, **kwargs):
+        super().__init__(prefix, **kwargs)
+
+        # if theta changes, update the pseudo position
+        self._theta0.subscribe(self.parameter_updated)
+        self.real_theta.subscribe(self.parameter_updated)
+
+    def parameter_updated(self, value=None, **kwargs):
+        self._update_position()
+
+    @property
+    def radian_theta(self):
+        return math.radians(self.real_theta.position + self._theta0.get())
```

Comment:
@tacaswell ponted the `.get()` should be used consistently:
```suggestion
        return math.radians(self.real_theta.get() + self._theta0.get())
```

Reply:
I changed this to .get() but then I was getting errors saying 'TypeError: can only concatenate tuple (not "float") to tuple.

If I'm tracing this correctly, self.real_theta returns a tuple with all the EpicsMotor information. If i change this to .get()[0], then it returns just the position and bluesky seems happy. Is this alright?